### PR TITLE
Adds TheVan RoadKit Gamepad Layout

### PR DESCRIPTION
**Pull request by wavebeem**
I didn't see any gamepad style layouts so I thought I would contribute mine

### CODE REVIEW

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/config.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/config.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_ortho_4x4(
+    KC_TAB,  KC_Q,    KC_W,    KC_E,
+    KC_ESC,  KC_A,    KC_S,    KC_D,
+    KC_LSFT, KC_Z,    KC_X,    KC_C,
+    KC_F,    KC_LALT, KC_LCTL, KC_SPC
+  ),
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c]: Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
+  switch (id) {
+    case 0:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL << 0);
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c]: Could you replace this function with `set_single_persistent_default_layer` (and remove the `1UL <<` part)

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    KC_F,    KC_LALT, KC_LCTL, KC_SPC
+  ),
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/keymap.c]: Could you convert this over to `process_record_user, as we have depreciated `action_get_macro`? 

It should be a pretty direct port.

https://docs.qmk.fm/#/feature_macros

File: keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/thevankeyboards/roadkit/keymaps/wavebeem-gamepad/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

**Comment by wavebeem** [comment]
Thanks for the advice @drashna. I copied another folder for the RoadKit that must've been outdated. Let me know if you see any issues with the changes I pushed up.

**Comment by drashna** [comment]
Yeah, we've been updating a lot of the boards with the newer stuff, but we haven't gotten to this one yet.  So that's why.

But thanks!  And nice!

**Pull request merged**